tests/ui/features/@sbom-explorer/sbom-explorer.step.ts

```diff
@@ -78,3 +81,77 @@ Then(
     await toolbarTable.verifyColumnContainsText(columnName, expectedValue);
   }
 );
+
+Given(
+  "An ingested {string} SBOM {string} containing Vulnerabilities",
+  async ({ page }, _sbomType, sbomName) => {
+    const searchPage = new SearchPage(page);
+    await searchPage.dedicatedSearch("SBOMs", sbomName);
+    const element = await page.locator(
+      `xpath=(//tr[contains(.,'${sbomName}')]/td[@data-label='Vulnerabilities']/div)[1]`
+    );
+    await expect(element, "SBOM have no vulnerabilities").toHaveText(
+      /^(?!0$).+/
+    );
+  }
+);
+
+When("User Clicks on Vulnerabilities Tab Action", async ({ page }) => {
+  await page.getByLabel("Tab action").click();
+});
+
+Then("Vulnerability Popup menu appears with message", async ({ page }) => {
+  await page.getByText("Any found vulnerabilities").isVisible();
+  await page.getByLabel("Close").click();
+});
+
+Then(
+  "Vulnerability Risk Profile circle should be visible",
+  async ({ page }) => {
+    await page.locator(`xpath=//div[contains(@class, 'chart')]`).isVisible();
+  }
+);
+
+Then(
+  "Vulnerability Risk Profile shows summary of vulnerabilities",
+  async ({ page }) => {
+    const detailsPage = new DetailsPage(page);
+    await detailsPage.verifyVulnerabilityPanelcount();
+  }
+);
+
+Then(
+  "SBOM Name {string} should be visible inside the tab",
+  async ({ page }, sbomName) => {
+    const panelSbomName = await page.locator(
+      `xpath=//section[@id='refVulnerabilitiesSection']//dt[contains(.,'Name')]/following-sibling::dd`
```

Comment:
I am just wondering if some of these assertions don't rely on the xpath too much. For example here. There is already an aria-label directly in the dd element and while these might take longer to find for Playwright, we can still narrow the search by a lot with the ID here.